### PR TITLE
fix: get talosconfig from secrets instead of talosconfig resources

### PR DESCRIPTION
**Pull request by Unix4ever**
This should address the issue when the management plane was migrated by
using `clusterctl move`, which drops all statuses and talosconfig
resources become empty.

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

**Review by smira — APPROVED** [review]

File: controllers/configs.go [hunk 1]
```diff
@@ -89,7 +90,24 @@ func (r *TalosControlPlaneReconciler) talosconfigForMachines(ctx context.Context
...
 	addrList := []string{}
 
-	var t *talosconfig.Config
+	var (
+		talosconfigSecret v1.Secret
+	)
+
+	if err := r.Client.Get(ctx,
+		types.NamespacedName{
+			Namespace: "default",
```
Comment (frezbo) [inline controllers/configs.go:99]: is this always the `default` namespace?

Comment (smira) [reply controllers/configs.go:99]: nice catch, it should be Machine's namespace (or Cluster's)

**Comment by DerTiedemann** [comment]
Thanks alot for the quick fix <3

**Pull request merged**